idn/punycode.go

```diff
@@ -52,6 +59,12 @@ func ToPunycode(s string) string {
 
 // FromPunycode returns unicode domain name from provided punycode string.
 func FromPunycode(s string) string {
+	// Early check to see if decoding is needed.
+	// This will prevent making heap allocations when not needed.
+	if !strings.Contains(s, _PREFIX) {
```

Comment:
[ Quoting <notifications@github.com> in "Re: [dns] Do simple prechecks to av..." ]
>> @@ -52,6 +59,12 @@ func ToPunycode(s string) string {
>>
>>  // FromPunycode returns unicode domain name from provided punycode string.
>>  func FromPunycode(s string) string {
>> +	// Early check to see if decoding is needed.
>> +	// This will prevent making heap allocations when not needed.
>> +	if !strings.Contains(s, _PREFIX) {
>
>That won't work since it can be in any of the labels. For example `www.xn--z-0ub.com`.

But I can have miekxn--miek as a label. If you want to do this properly you need
to Split and check if one of the labels start with the _PREFIX.

/Miek

--
Miek Gieben